src/vs/platform/terminal/common/environmentVariableCollection.ts

```diff
@@ -5,21 +5,31 @@
 
 import { IProcessEnvironment, isWindows } from 'vs/base/common/platform';
 import { EnvironmentVariableMutatorType, IEnvironmentVariableCollection, IExtensionOwnedEnvironmentVariableMutator, IMergedEnvironmentVariableCollection, IMergedEnvironmentVariableCollectionDiff } from 'vs/platform/terminal/common/environmentVariable';
+import { IWorkspaceFolder } from 'vs/platform/workspace/common/workspace';
 
 type VariableResolver = (str: string) => Promise<string>;
 
 export class MergedEnvironmentVariableCollection implements IMergedEnvironmentVariableCollection {
 	readonly map: Map<string, IExtensionOwnedEnvironmentVariableMutator[]> = new Map();
 
 	constructor(
-		readonly collections: ReadonlyMap<string, IEnvironmentVariableCollection>
+		readonly collections: ReadonlyMap<string, IEnvironmentVariableCollection>,
+		readonly workspaceFolder?: IWorkspaceFolder
```

Comment:
TODO: Change it to `readonly workspaceFolder: IWorkspaceFolder | undefined` once optional (`?`) is removed.

Reply:
`MergedEnvironmentVariableCollection` represents a Terminal, and each Terminal is created for a specific workspace folder. So I think it's right as it is.

Reply:
Oh my bad, I misunderstood. Can we remove `readonly` (which implies `public`) from this and rename it to `owningWorkspace` to make the intent more clear?